tests/unittest.py

```diff
@@ -334,6 +335,12 @@ def setUp(self) -> None:
         calling the prepare function.
         """
         self.reactor, self.clock = get_clock()
+
+        # The home server will start connecting places as soon as it starts,
+        # so we must update some fake DNS entries early.
```

Comment:
```suggestion
        # The homeserver will start connecting places as soon as it starts,
        # so we must update some fake DNS entries early.
```

What connects immediately? Couldn't we do this in the `setUp` of the email tests and simply call `self.reactor.lookups[...] = ...`?

Reply:
So, the context for that comment is:

- it was mostly redacted with #11508, which in retrospective was too ambitious and not a right path to contribute to Synapse (incrementally and with actual bug fixing instead of just tests)

So, for example: `BaseMultiWorkerStreamTestCase` forces `self.reactor.lookups` in its `setUp`, as well, and it makes a point that there might be already some Redis connections by the time that part of the code hits.

I'd leave this up to you, we could move the setting of the reactor's lookups to the email test, the logic for doing it here was that other tests could start taking advantage of this mechanism over time.

Will mark as resolved and if you deem it necessary please open it again and we'll deal with it :-).